File: BOJ/1002/1002.py
#!/usr/bin/env python3

from math import sqrt

def calculate(x1, y1, r1, x2, y2, r2):
    dist = sqrt((x1 - x2) ** 2 + (y1 - y2) ** 2)
    max_r = max(r1, r2)
    min_r = min(r1, r2)

    # When the center are same
    if dist == 0:
        # r1 == r2, it's same circle
        if max_r == min_r:
            return -1
        # otherwise, no any point intersects
        else:
            return 0
    # When the two circles tangents
    elif dist == (max_r + min_r) or dist == (max_r - min_r):
        return 1
    # When the two circle doesn't intersects
    elif dist > (max_r + min_r) or max_r > (dist + min_r):
        return 0
    else:
        return 2

if __name__ == '__main__':
    cases = int(input())

    for i in range(0, cases):
        x1, y1, r1, x2, y2, r2 = map(int, input().split())
        answer = calculate(x1, y1, r1, x2, y2, r2)
        print(answer)

File: PS/BOJ/1002/1002.cc
#include <iostream>
#include <cmath>
using namespace std;

typedef struct _turret {
	int x;
	int y;
	int radius;
} turret;

int calculate(turret t1, turret t2) {
	int diffx = t1.x - t2.x;
	int diffy = t1.y - t2.y;
	float dist = sqrt(diffx * diffx + diffy * diffy);

	int max_r = (t1.radius > t2.radius) ? t1.radius : t2.radius;
	int min_r = (t1.radius < t2.radius) ? t1.radius : t2.radius;

	// when two turrets are in same position
	if (dist == 0) {
		if (t1.radius == t2.radius) {
			return -1;
		}
		else 
			return 0;
	}
	// when two turrets are tangent
	else if (dist == max_r + min_r || dist == max_r - min_r) {
		return 1;
	}
	// when two turrets are not meet
	else if (dist > max_r + min_r || max_r > dist + min_r) {
		return 0;
	}
	else
		return 2;
}

int main(void) {
	int cases;

	cin >> cases;
	for (int i = 0; i < cases; i++) {
		turret t1, t2;
		cin >> t1.x >> t1.y >> t1.radius
			>> t2.x >> t2.y >> t2.radius;
		cout << calculate(t1, t2) << '\n';
	}
}